src/Agent.Listener/JobDispatcher.cs

```diff
@@ -88,6 +88,18 @@ public void Run(Pipelines.AgentJobRequestMessage jobRequestMessage, bool runOnce
                 }
             }
 
+            var service = HostContext.GetService<IFeatureFlagProvider>();
+            string ffState;
+            try
+            {
+                ffState = service.GetFeatureFlagAsync(HostContext, "DistributedTask.Agent.EnableAdditionalMaskingRegexes", Trace)?.Result?.EffectiveState;
```

Comment:
Are we expecting null here?

Reply:
For another PR, it would be nice if we had a wrapper that returned a bool (EffectiveState only) so the caller doesn't need to do the null checking or compare a string.

Reply:
@merlynomsft  I will modify provider to have more useful experience in the separate PR